src/main/java/com/android/volley/toolbox/HurlStack.java

```diff
@@ -168,10 +170,25 @@ public void close() throws IOException {
         }
     }
 
+    /**
+     * Overload this method for manipulate response stream (Use 'write (byte[] b, int off, int len)' in subclasse).
+     *
+     * @param request current request.
+     * @param connection current connection of request.
+     * @param responseCode response code.
+     * @param length size of response stream.
+     * @return an UrlConnectionInputStream object for read response.
+     */
+    protected FilterInputStream createInputStream (Request<?> request, HttpURLConnection connection,
```

Comment:
the return type here should be InputStream, not FilterInputStream. Volley doesn't care/depend on what kind of InputStream gets returned here - any InputStream is fine. The fact that we use FilterInputStream here and in your (presumed) subclass isn't a requirement, just an implementation detail.